kapitan/targets.py

```diff
@@ -414,14 +415,16 @@ def compile_target(target_obj, search_paths, compile_path, ref_controller, inven
     ext_vars = target_obj["vars"]
     target_name = ext_vars["target"]
 
-    jinja2_compiler = Jinja2(compile_path, search_paths, ref_controller, inventory_path)
-    jsonnet_compiler = Jsonnet(compile_path, search_paths, ref_controller)
-    kadet_compiler = Kadet(compile_path, search_paths, ref_controller)
-    helm_compiler = Helm(compile_path, search_paths, ref_controller)
-    copy_compiler = Copy(compile_path, search_paths, ref_controller)
-    remove_compiler = Remove(compile_path, search_paths, ref_controller)
-
     for comp_obj in compile_objs:
+        # these need to be inside this for loop so default values are reinitialized for each compile object
```

Comment:
OK. I feel that at some point, these should be instantiated on demand.

Reply:
done in this PR